### PR TITLE
Collapse merge maneuvers

### PR DESCRIPTION
**Pull request by dgearhart**
# Issue

Merging onto highways is an obvious maneuver with no choice, therefore, we can collapse the merge maneuver.

---

## Example
image

``` diff
< BEFORE
< 5: Turn left to take the PA 283 West ramp toward Harrisburg/Harrisburg International Airport. | 0.4 mi
< 6: Merge left onto PA 283 West. | 3.8 mi
< 7: Take the I 76 Toll/PATP Toll exit on the left toward Philadelphia/Pittsburgh. | 0.1 mi
< 8: Merge left onto I 283 South. | 0.1 mi
< 9: Keep right to take I 76 West toward Pittsburgh. | 0.5 mi
< 10: Merge left onto I 76 West/PATP West/Pennsylvania Turnpike. | 4.6 mi
< 11: Take exit 242 onto I 83 toward York/Baltimore. | 0.7 mi
< 12: Take the I 83 South ramp on the left toward York/Baltimore. | 0.4 mi
< 13: Merge left onto I 83 South. | 61.8 mi
< 14: Keep right to take I 695 West toward Baltimore. | 0.5 mi
< 15: Merge left onto I 695 West/Baltimore Beltway. | 20.7 mi
< 16: Take exit 4 on the left onto I 97 South toward Annapolis/Bay Bridge. | 0.8 mi
< 17: Merge left onto I 97 South. | 7.4 mi
< 18: Take exit 10A onto Benfield Boulevard East toward Veterans Highway/Severna Park. | 0.4 mi
< 19: Merge left onto Benfield Boulevard. | 1.3 mi
< 20: Turn left onto St Ives Drive. | 0.6 mi
> AFTER
> 5: Turn left to take the PA 283 West ramp toward Harrisburg/Harrisburg International Airport. | 4.2 mi
> 6: Take the I 76 Toll/PATP Toll exit on the left toward Philadelphia/Pittsburgh. | 0.2 mi
> 7: Keep right to take I 76 West toward Pittsburgh. | 5.0 mi
> 8: Take exit 242 onto I 83 toward York/Baltimore. | 0.7 mi
> 9: Take the I 83 South ramp on the left toward York/Baltimore. | 62.1 mi
> 10: Keep right to take I 695 West toward Baltimore. | 21.1 mi
> 11: Take exit 4 on the left onto I 97 South toward Annapolis/Bay Bridge. | 8.3 mi
> 12: Take exit 10A onto Benfield Boulevard East toward Veterans Highway/Severna Park. | 1.7 mi
> 13: Turn left onto St Ives Drive. | 0.6 mi
```
---

## This improves `50%` of user routes

---

## Tasklist

 - [x] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

File: test/gurka/test_instructions_collapse_merge_toward.cc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  // Verify left fork and collapsed merge
+  gurka::assert::raw::
+      expect_instructions_at_maneuver_index(result, maneuver_index,
+                                            "Keep left to take A6 toward Harrisburg/Lancaster.",
+                                            "Keep left to take A6.",
+                                            "Keep left to take A6 toward Harrisburg, Lancaster.",
+                                            "Continue for 6 kilometers.");
+
+  // Collapsed
```
Comment (gknisely) [inline test/gurka/test_instructions_collapse_merge_toward.cc:78]: @dgearhart is this just leftover

File: test/gurka/test_instructions_collapse_merge.cc [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  // Keep right with collapsed merge
+  gurka::assert::raw::
+      expect_instructions_at_maneuver_index(result, maneuver_index,
+                                            "Keep right to take A 70 East toward Baltimore.",
+                                            "Keep right to take A 70 East.",
+                                            "Keep right to take A 70 East toward Baltimore.",
+                                            "Continue for 6 kilometers.");
+
+  // Collapsed merge info
+  //  ++maneuver_index;
```
Comment (gknisely) [inline test/gurka/test_instructions_collapse_merge.cc:211]: @dgearhart is this just leftover

File: test/gurka/test_instructions_collapse_merge.cc [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  gurka::assert::raw::
+      expect_instructions_at_maneuver_index(result, maneuver_index,
+                                            "Keep left to take A 76/Pennsylvania Turnpike.",
+                                            "Keep left to take A 76.",
+                                            "Keep left to take A 76, Pennsylvania Turnpike.",
+                                            "Continue for 8 kilometers.");
+  // Collapsed merge info
+  //  ++maneuver_index;
+  //
+  //  // Merge left
```
Comment (gknisely) [inline test/gurka/test_instructions_collapse_merge.cc:172]: @dgearhart is this just leftover

**Comment by gknisely** [comment]
@dgearhart few tests with commented out tests/logic.  I did not comment on them all.  Was that intentional?

**Comment by dgearhart** [comment]
> @dgearhart few tests with commented out tests/logic. I did not comment on them all. Was that intentional?

@gknisely yes, kept for now so we had history

**Review by gknisely — APPROVED** [review]

**Review by danpaz — DISMISSED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -227,11 +227,12 @@
    * ADDED: Add Estonian locale. [#2748](https://github.com/valhalla/valhalla/pull/2748)
    * CHANGED: Handle GraphTile objects as smart pointers [#2703](https://github.com/valhalla/valhalla/pull/2703)
    * CHANGED: Improve stability with no RTTI build [#2759](https://github.com/valhalla/valhalla/pull/2759) and [#2760](https://github.com/valhalla/valhalla/pull/2760)
-   * CHANGED: Change generic service roads to a new Use=kServiceRoad. This is for highway=service without other service=* tags (such as driveway, alley, parking aisle) [#2419](https://github.com/valhalla/valhalla/pull/2419)
+   * CHANGED: Change generic service roads to a new Use=kServiceRoad. This is for highway=service without other service= tags (such as driveway, alley, parking aisle) [#2419](https://github.com/valhalla/valhalla/pull/2419)
```
Comment (mandeepsandhu) [inline CHANGELOG.md:230]: :slightly_smiling_face: the `*` in that line messing up formatting for the subsequent lines (in the rendered markdown) has been bothering me as well! :+1:

**Review by danpaz — DISMISSED** [review]

**Review by mandeepsandhu — APPROVED** [review]

**Comment by lxbarth** [comment]
w00t!

**Comment by chrisrieth** [comment]
@kshehadeh @ariel - per our conversation in auto standup, TME is now reporting missing merge instructions as a regression:

https://github.com/mapbox/customer-evaluation-issues/issues/137#issuecomment-779504168 
https://github.com/mapbox/customer-evaluation-issues/issues/150#issuecomment-779848150 
https://github.com/mapbox/customer-evaluation-issues/issues/161#issuecomment-779845445

**Comment by ArielSmoliar** [comment]
We will need to find a way to communicate improvement metrics.

**Comment by ariel** [comment]
Wrong Ariel.

**Pull request merged**